django/db/models/enums.py

```diff
@@ -0,0 +1,64 @@
+import enum
+
+from django.core import exceptions
+from django.utils.translation import gettext_lazy as _
+
+__all__ = ['ChoiceIntEnum', 'ChoiceStrEnum']
+
+
+class ChoiceEnumMeta(enum.EnumMeta):
+    """A metaclass for creating a enum choices."""
+
+    def __new__(metacls, classname, bases, classdict):
+        choices = {}
+        for key in classdict._member_names:
+            value = classdict[key]
+            if isinstance(value, (list, tuple)):
+                try:
+                    value, label = value
+                except ValueError as e:
+                    raise e.__class__(f'Invalid member for {classname}: {key}.') from e
```

Comment:
Please don't use f-string, for now we don't allow them in Django's codebase.

Is there any reason to use `e.__class__` instead of `ValueError`?

Reply:
> Please don't use f-string, for now we don't allow them in Django's codebase.

Sorry. I misunderstood https://github.com/django/django/pull/11223#discussion_r279075913.

> Is there any reason to use `e.__class__` instead of `ValueError`?

I guess not in this case as we're only catching one type of exception. Sometimes it is useful if you want to keep the same exception class but change the message as in this case. Will change.